go-controller/pkg/ovn/namespace.go

```diff
@@ -67,11 +70,38 @@ func (oc *DefaultNetworkController) syncNamespaces(namespaces []interface{}) err
 		expectedNs[ns.Name] = true
 	}
 
-	err := oc.addressSetFactory.ProcessEachAddressSet(func(addrSetName, namespaceName, nameSuffix string) error {
-		if nameSuffix == "" && !expectedNs[namespaceName] {
-			if err := oc.addressSetFactory.DestroyAddressSetInBackingStore(addrSetName); err != nil {
-				klog.Errorf(err.Error())
-				return err
+	err := oc.addressSetFactory.ProcessEachAddressSet(func(hashedName, addrSetName string) error {
```

Comment:
this is a bit hard to read. I wonder if we could at least reduce the nesting on it by simplifying the if condition to something like:
```
if strings.HasPrefix(addrSetName, types.HybridRoutePolicyPrefix) ||
   strings.HasPrefix(addrSetName, types.EgressQoSRulePrefix) ||
   strings.Contains(addrSetName, ".") {
   return nil
}
 ... <resume> one ident level less than before ...
```